scripts/spell-check.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/sh
+#
+## Copyright (C) 2020 The Squid Software Foundation and contributors
+##
+## Squid software is distributed under GPLv2+ license and includes
+## contributions from numerous individuals and organizations.
+## Please see the COPYING and CONTRIBUTORS files for details.
+##
+
+#
+# This script runs codespell against selected files.
+#
+# Use -q or --quiet option to run quietly
+#
+
+CODESPELL_LOC=`which codespell`
+if test "${CODESPELL_LOC}" = ""; then
+    echo "This script requires codespell which was not found."
+    exit 1
+fi
+
+CODESPELL_VER=`codespell --version 2>&1`
+echo "The codespell version is ${CODESPELL_VER}."
```

Comment:
I think we can simplify this quite a bit and avoid relying on "which". Something along these lines, perhaps?

```suggestion
echo -n "Codespell version: "
if ! codespell --version; then
    echo "This script requires codespell which was not found."
    exit 1
fi
```

Reply:
Applied in next commit.